Parsers/ASimWebSession/Parsers/vimWebSessionBarracudaWAF.yaml

```diff
@@ -0,0 +1,194 @@
+  Version: '0.2.6'
+References:
+- Title: ASIM Web Session Schema
+  Link: https://aka.ms/ASimWebSessionDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+- Title: Logs type Description in Barracuda WAF
+  Link: https://campus.barracuda.com/product/webapplicationfirewall/doc/78810008/types-of-logs/
+Description: |
+  This ASIM parser supports filtering and normalizing Barracuda WAF logs to the ASIM Web Session normalized schema.
+ParserName: vimWebSessionBarracudaWAF
+EquivalentBuiltInParser: _Im_WebSession_BarracudaWAF
+ParserParams:
+  - Name: starttime
+    Type: datetime
+    Default: datetime(null)
+  - Name: endtime
+    Type: datetime
+    Default: datetime(null)
+  - Name: srcipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: ipaddr_has_any_prefix
+    Type: dynamic
+    Default: dynamic([])
+  - Name: url_has_any
+    Type: dynamic
+    Default: dynamic([])
+  - Name: httpuseragent_has_any
+    Type: dynamic 
+    Default: dynamic([])
+  - Name: eventresult
+    Type: string
+    Default: '*'
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let SeverityLookup = datatable (severity: int, EventSeverity: string)
+      [
+      0, "High", 
+      1, "High", 
+      2, "High", 
+      3, "Medium",
+      4, "Low",
+      5, "Low", 
+      6, "Informational",
+      7, "Informational" 
+  ];
+  let EventResultWFLookup = datatable (
+      Action_s: string,
+      EventResult_WF: string,
+      DvcAction: string
+  )
+      [
+      "LOG", "Success", "ALLOW",
+      "DENY", "Failure", "DENY",
+      "WARNING", "Success", "WARNING"
+  ];
+  let EventTypeLookup = datatable (LogType_s: string, EventType: string, EventOriginalType: string )
+      [
+      "WF", "HTTPsession", "Web Firewall",
+      "TR", "WebServerSession", "Access"
+  ];
+  let parser=(
+      starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]), 
+      ipaddr_has_any_prefix: dynamic=dynamic([]), 
+      url_has_any: dynamic=dynamic([]),
+      httpuseragent_has_any: dynamic=dynamic([]),
+      eventresult: string='*',
+      disabled: bool=false
+      ) {
+      let src_or_any=set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      barracuda_CL
+      | where not(disabled) and (LogType_s == "WF" or LogType_s == "TR")
+      | extend EventStartTime = unixtime_seconds_todatetime(tolong(DeviceReceiptTime_s))
+      | extend EventEndTime = EventStartTime
+      | where (isnull(starttime) or EventStartTime >= starttime)
+          and (isnull(endtime) or EventEndTime <= endtime)
+      | where (array_length(url_has_any) == 0 or URL_s has_any (url_has_any))
+      | where (array_length(httpuseragent_has_any) == 0 or UserAgent_s has_any(httpuseragent_has_any))
+      | extend
+          temp_SrcMatch=has_any_ipv4_prefix(ClientIP_s, src_or_any),
+          temp_DstMatch=has_any_ipv4_prefix(ServerIP_s, ipaddr_has_any_prefix)
```

Comment:
Column_ifexists is missing for these.

Reply:
Instead of column_ifexists we used the the approach suggested by you in mail to create one data table with the default values.